server/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -279,10 +285,19 @@ Result prepareUpdateScriptRequest(ShardId shardId, UpdateRequest request, GetRes
     private Map<String, Object> executeScript(Script script, Map<String, Object> ctx) {
         try {
             if (scriptService != null) {
-                ExecutableScript.Factory factory = scriptService.compile(script, ExecutableScript.UPDATE_CONTEXT);
-                ExecutableScript executableScript = factory.newInstance(script.getParams());
-                executableScript.setNextVar(ContextFields.CTX, ctx);
-                executableScript.run();
+                UpdateScript.Factory factory = scriptService.compile(script, UpdateScript.CONTEXT);
+                final Map<String, Object> params;
+                if (CTX_IN_PARAMS) {
+                    params = new HashMap<>();
+                    params.putAll(script.getParams());
+                    params.put(ContextFields.CTX, ctx);
+                    deprecationLogger.deprecated("Exposing `ctx` under `params.ctx` is deprecated. " +
```

Comment:
nit: "Exposing `ctx` under" -> "Using `ctx` via"

Also, I would use "to enforce the non-deprecated usage" instead of "to disable this behavior"